spring/src/main/java/io/cucumber/spring/SpringFactory.java

```diff
@@ -129,21 +129,24 @@ private static boolean hasAnnotation(Annotation annotation, Collection<Class<? e
 
     @Override
     public void start() {
-        if (stepClassWithSpringContext != null) {
-            testContextManager = new CucumberTestContextManager(stepClassWithSpringContext);
-        } else {
-            if (beanFactory == null) {
-                beanFactory = createFallbackContext();
+        synchronized (monitor) {
```

Comment:
Replaced the `SpringFactory.class` with `monitor`. The class is accessible by any one. And while I don't expect any one to use it to synchronize on it, there also no need to do so.

Reply:
Also explained the why of it all.